scripts/version_buckets.py

```diff
@@ -18,9 +18,24 @@
 buckets = s3.list_buckets()
 for bucket_dict in buckets["Buckets"]:
     bucket_name = bucket_dict["Name"]
-    if not bucket_name.startswith("alpha"):
-        # Only deal with buckets created and used by/for the control panel
-        # (always prefixed with "alpha").
+    bucket = boto3.resource("s3").Bucket(bucket_name)
+    # Pre-flight checks for the bucket. Only buckets created by the control
+    # panel should be updated. They either start with the name "alpha-app-" or
+    # they have a tag of "buckettype" with the value "datawarehouse".
+    # See discussion here:
+    # https://github.com/ministryofjustice/analytics-platform-ops/pull/320#discussion_r433765261
+    update_flag = bucket_name.startswith("alpha-app-")
+    try:
+        tags = bucket.Tagging()
+        tags.load()
+        for obj in tags.tag_set:
+            if obj["Key"] == "buckettype" and obj["Value"] == "datawarehouse":
+                update_flag = True
+                break
+    except:
```

Comment:
A bare except for any particular reason? Are you just wanting to catch KeyError or are there others?

Reply:
Good question. This was a matter of convenience. The bucket tagging is complicated, inconsistent and so there were several exceptions that could have been thrown. At this point in the code the only thing that is interesting is to set the flag if, and only if, the conditions on line 32 are met. For all other situations (including throwing an exception) the bucket can be safely ignored.

Reply:
A couple of points from PEP8

> A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except Exception: (bare except is equivalent to except BaseException:).

How about `except Exception:` as a minimum, otherwise at this time a Ctrl-C will cause the script to ignore a bucket but carry on running.

> Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs

What exceptions will `tags = bucket.Tagging()` raise if there is a random network error in the AWS call? In this case we would want to know about it, rather than ignore the bucket.

Reply:
I can attest that it is annoying not to be able to use Ctrl-C during `python version_buckets.py list`